### PR TITLE
Enable configuration management via viper

### PR DESCRIPTION
**Pull request by supreeth7**
### What type of PR is this?
Feature

### What this PR does / Why we need it?
- This PR leverages [viper](https://github.com/spf13/viper) to enable effective configuration management for Backplane CLI.
- The configuration proxy and Backplane URL set by the users via the `HTTPS_PROXY` and `BACKPLANE_URL` environment variables are respected and given preference over the values defined in the configuration file which wasn't the case before.

### Which Jira/Github issue(s) does this PR fix?

Resolves: [OSD-15922](https://issues.redhat.com/browse/OSD-15922)

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [x] Included documentation changes with PR

### CODE REVIEW

**Comment by wanghaoran1988** [comment]
```
{
	"proxy-url": "${URL}",
	"ocm-prod": "${URL}",
	"ocm-stg":"${URL}",
	"ocm-int":"${URL}",
	"bp-url-prod": "${URL}",
	"bp-url-stg":"${URL}",
	"bp-url-int":"${URL}"
}
```
This is a breaking change for the cli config,  I don't think it's a good option to put all those urls in this config.

**Comment by wanghaoran1988** [comment]
> > I don't think it's a good option to put all those urls in this config
> 
> Thanks for taking a look, I understand, but what is stopping us from having a single config file? Is it a security concern? Anyhow we are exposing URLs in the current config, with this we don't have to switch between multiple config files until we have a mapping defined in OCM API.
Single config means you are more easy to make mistake, expose different env remind you which env you are accessing.

**Comment by supreeth7** [comment]
The vulnerabilities outweigh the use case.
Let us have SDA incorporate this on the OCM side where we have a field in the provision shard object that is the `backplane_url` where SDA can populate that in the `environment` DB table exposing it in the cluster payload.

**Comment by supreeth7** [comment]
@wanghaoran1988 I've removed the explicit mapping which doesn't break the CLI. This PR integrates viper for config management please take a look.
This work will be followed by https://issues.redhat.com/browse/OSD-15878

File: pkg/cli/config/config.go [hunk 1]
```diff
@@ -1,35 +1,21 @@
...
 	"path/filepath"
 
 	"github.com/openshift/backplane-cli/pkg/info"
+	"github.com/spf13/viper"
 )
 
 type BackplaneConfiguration struct {
-	URL      string `json:"url"`
-	ProxyURL string `json:"proxy-url"`
+	URL      string `mapstructure:"url"`
```
Comment (wanghaoran1988) [inline pkg/cli/config/config.go:13]: Could you help me understand why this?

Comment (supreeth7) [reply pkg/cli/config/config.go:13]: From the [documentation](https://github.com/spf13/viper#unmarshaling):

> Viper uses [github.com/mitchellh/mapstructure](https://github.com/mitchellh/mapstructure) under the hood for unmarshaling values which uses mapstructure tags by default.

File: pkg/cli/config/config.go [hunk 2]
```diff
@@ -40,51 +26,54 @@ func getDefaultConfigPath() (string, error) {
...
-			}
+		if err := viper.Unmarshal(&bpConfig); err != nil {
+			return bpConfig, err
+		}
+	}
 
-			return bpConfig, nil
+	// Check if user has explicitly defined backplane URL
+	// It has higher precedence over the config file
+	bpURL, hasURL := os.LookupEnv(info.BACKPLANE_URL_ENV_NAME)
```
Comment (wanghaoran1988) [inline pkg/cli/config/config.go:62]: viper does support read configs considering environment setup, could you try to see how to integrate this? https://github.com/spf13/viper#working-with-environment-variables

Comment (wanghaoran1988) [reply pkg/cli/config/config.go:62]: I see no value use viper without using this feature.

Comment (supreeth7) [reply pkg/cli/config/config.go]: I had this added in another branch, moved it here, thanks

**Pull request merged**